syntax = "proto3";
// The greeting service definition.
service HierarchyService {
  // Sends a greeting
  rpc Query (NodeList) returns (NodeList) {}
}

message Node{
  map<string, Property> map = 1;
}

message IntList {
    repeated int64 values = 1;
}

message StringList  {
    repeated string values = 1;
}

message NodeList {
    repeated Node node = 1;
}

message Property {
  oneof value {
      bool boolean = 1;
      int64 integer = 2;
      float floating = 3;
      string stringval = 4;
      IntList intlist = 5;
      StringList stringlist = 6;
  };
}